### DISCUSSION TITLE
ERROR (Kamal::Cli::Build::BuildError): Docker buildx plugin is not installed locally

### DISCUSSION THREAD

**Discussion by ItaloCobains** [basecamp/kamal#726] [General]
How to resolve this ?

![image](https://github.com/basecamp/kamal/assets/83548040/77dd249f-422d-48dc-a661-6ac521088e49)

**Comment by ItaloCobains** [comment]
I didn't correctly interpret the error, sorry.

`sudo apt install docker-buildx` resolve.

**Reply by bufordtaylor** [reply]
Shouldn't this be something kamal handles in the `setup` script?

**Reply by tom-mullen** [reply]
This error is relating to the local version (i.e you dev machine) as opposed to the remote box. In order to get past the error, you need to make sure `docker buildx` runs locally.

Now, if you're on a mac and using homebrew, you need to make sure that you have both `docker` installed as well as the plugin `docker-buildx`.

Crucially, you need to make sure the plugin is discoverable by adding the following to the  `~/.docker/config.json`:

```
  "cliPluginsExtraDirs": [
    "/opt/homebrew/lib/docker/cli-plugins"
  ]
```

Once you have done that `docker buildx version` will run as expected and you should be on your way.
